tests/datasource/fluent/test_pandas_datasource.py

```diff
@@ -476,6 +476,27 @@ def test_dataframe_asset(
     )
 
 
+@pytest.mark.cloud
+def test_cloud_add_dataframe_asset(test_df_pandas: pd.DataFrame):
+    context = gx.get_context(mode="cloud")
+    datasource_name = "pandas_debugging_datasource"
+    dataframe_asset_name = "test_df"
+    try:
+        datasource = context.sources.add_pandas(name=datasource_name)
+    except gx.exceptions.exceptions.DataContextError:
+        datasource = context.get_datasource(datasource_name=datasource_name)  # type: ignore[assignment]
+    try:
+        dataframe_asset = datasource.add_dataframe_asset(name=dataframe_asset_name)
+    except ValueError:
+        dataframe_asset = datasource.get_asset(asset_name=dataframe_asset_name)
+
+    dataframe_asset.build_batch_request(dataframe=test_df_pandas)
+
+    assert dataframe_asset.dataframe.equals(
+        context.datasources[datasource_name].assets[0].dataframe  # type: ignore[union-attr]
+    )
+
```

Comment:
Is this an E2E test? Can we avoid `try/except` logic and instead set up guaranteed prereq state?

Reply:
Updated to use just the `get` commands.